src/analyzer/protocol/smb/smb2-com-read.pac

```diff
@@ -55,6 +55,13 @@ refine connection SMB_Conn += {
 			                 bro_analyzer()->Conn(), h->is_orig());
 			}
 
+		if ( smb2_read_request )
+			{
+			BifEvent::generate_smb2_read_response(bro_analyzer(),
```

Comment:
You are generating a read_response message but checking above for a read_request.  You should be checking for the same event that is being generated in the if statement.

Reply:
correct